gsy_framework/sim_results/electric_blue/volume_timeseries.py

```diff
@@ -0,0 +1,160 @@
+from typing import Dict, Optional
+
+from pendulum import DateTime
+
+from gsy_framework.enums import AggregationResolution, AvailableMarketTypes
+from gsy_framework.forward_markets.aggregated_ssp import get_aggregated_SSP
+from gsy_framework.forward_markets.forward_profile import (
+    ForwardTradeProfileGenerator)
+from gsy_framework.sim_results.electric_blue.aggregate_results import (
+    ForwardDeviceStats)
+
+FORWARD_MARKET_TYPES = [
+    AvailableMarketTypes.YEAR_FORWARD, AvailableMarketTypes.MONTH_FORWARD,
+    AvailableMarketTypes.WEEK_FORWARD, AvailableMarketTypes.DAY_FORWARD,
+]
+
+# a dictionary showing the start of time slot for each resolution.
+START_OF = {
+    AggregationResolution.RES_1_HOUR: "hour",
+    AggregationResolution.RES_1_WEEK: "week",
+    AggregationResolution.RES_1_MONTH: "month",
+    AggregationResolution.RES_1_YEAR: "year",
+}
+
+
+class AssetVolumeTimeSeries:
+    """This class generates combined volume time series for the whole year for each asset.
+    The result would be like this for a monthly resolution for the year 2020:
+    {
+        DateTime(2020, 1, 1, 0, 0, 0): {
+            DateTime(2020, 1, 1, 0, 0, 0): {
+                "SSP": 1005.4160297565659,
+                "YEAR_FORWARD": {
+                    "bought_kWh": 201.08320595131318,
+                    "sold_kWh": 201.08320595131318,
+                },
+                "MONTH_FORWARD": {
+                    "bought_kWh": 201.08320595131318,
+                    "sold_kWh": 201.08320595131318,
+                },
+                "WEEK_FORWARD": {
+                    "bought_kWh": 32.432775153437625,
+                    "sold_kWh": 32.432775153437625,
+                },
+                "DAY_FORWARD": {
+                    "bought_kWh": 6.486555030687522,
+                    "sold_kWh": 6.486555030687522,
+                },
+            },
+            DateTime(2020, 2, 1, 0, 0, 0): {...},
+            ...
+            DateTime(2020, 12, 1, 0, 0, 0): {...},
+        },
+    }
+    """
+    def __init__(
+            self, asset_uuid: str, asset_peak_kWh: float,
+            resolution: AggregationResolution):
+
+        self.asset_uuid = asset_uuid
+        self.asset_peak_kWh = asset_peak_kWh
+        self.resolution = resolution
+
+        self._asset_volume_time_series_buffer: Dict[DateTime, Dict] = {}
+        self._trade_profile_generator = ForwardTradeProfileGenerator(self.asset_peak_kWh)
+
+    def add(self, asset_stats: ForwardDeviceStats, market_type: AvailableMarketTypes):
```

Comment:
The there is an instance of `ForwardDeviceStats` for each 15 min time slot, right?

Reply:
Also: Here you use `market_type` (which I prefer) but it was decided to use `product_type` right?

Reply:
Last but not least: How about rename this method to be more descriptive: e.g. `add_results_to_time_series` or `add_trade_to_time_series`?

Reply:
- There's an instance of the `ForwardDeviceStats` every 15 minutes but only ones that have new trades are taken into account.
- I think `market_type` is as valid as `product_type` as the enum itself is called: `AvailableMarketTypes`. What do you think? Should we change it to product_type anyway? 
- I'll rename it to `update_volume_time_series`.

Reply:
> I think market_type is as valid as product_type as the enum itself is called: AvailableMarketTypes. What do you think? Should we change it to product_type anyway

IMO `product_type` is not a good naming. But now as it was decided to used this name, lets be consistent throughout the code. Indeed, this means als to rename `AvailableMarketTypes`. This all can be done in a refactoring PR IMO.

Reply:
Fixed.